genic/main.c

```diff
@@ -117,15 +126,26 @@ int main(int argc, char **argv)
 
   /*Now make the gas if required*/
   if(All2.ProduceGas) {
-    setup_grid(shift_gas, All2.Ngrid);
+
+    if(!All2.MakeGlass) {
+        setup_grid(shift_gas, All2.Ngrid);
+    } else {
+        setup_glass(shift_gas, All2.Ngrid, (All2.Seed + 1) * 99999721);
```

Comment:
They are independent glasses.  I wanted to avoid the situation where `every CDM has a nearby baryon` in the initial condition. Reusing the same glass grid means any residual clustering due to the glass will be correlated at all scales between CDM and Baryon.

Using independent glass grids the residual clustering of gas and CDM are uncorrelated.  An interesting alternative is to use a single giant glass grid for all species, then randomly color particles as different species.

Reply:
Ah! That is very clever.